amethyst_renderer/examples/material.rs

```diff
@@ -103,11 +101,14 @@ fn main() {
             _ => (),
         });
 
-        renderer.draw(&scene, &pipe, delta);
+        renderer.draw(&scene, &pipe, delta)?;
         delta = Instant::now() - start;
     }
+    Ok(())
 }
 
+quick_main!(run);
```

Comment:
Examples should, if possible, avoid macros IMO.

Reply:
[quick_main!](https://github.com/rust-lang-nursery/error-chain/blob/master/src/quick_main.rs) could easily be replaced with how it is recommended to be done: [link](https://github.com/rust-lang-nursery/error-chain/blob/master/examples/quickstart.rs)


'''
fn main() {
    if let Err(ref e) = run() {
        use std::io::Write;
        let stderr = &mut ::std::io::stderr();
        let errmsg = "Error writing to stderr";

        writeln!(stderr, "error: {}", e).expect(errmsg);

        for e in e.iter().skip(1) {
            writeln!(stderr, "caused by: {}", e).expect(errmsg);
        }

        // The backtrace is not always generated. Try to run this example
        // with `RUST_BACKTRACE=1`.
        if let Some(backtrace) = e.backtrace() {
            writeln!(stderr, "backtrace: {:?}", backtrace).expect(errmsg);
        }

        ::std::process::exit(1);
    }
}
'''